weave/trace_server_bindings/remote_http_trace_server.py

```diff
@@ -123,6 +125,68 @@ def from_env(cls, should_batch: bool = False) -> "RemoteHTTPTraceServer":
     def set_auth(self, auth: t.Tuple[str, str]) -> None:
         self._auth = auth
 
+    def _strip_large_values_from_batch(
+        self, batch: t.List, encoded_bytes: int
+    ) -> t.List:
+        """
+        Iterate through the batch and replace large values with placeholders.
+
+        If keys on inputs or output on start/end reqs are larger than 1MiB,
+        replace them with placeholder values.
+        """
+
+        def _num_bytes(data: t.Any) -> int:
+            """
+            Calculate the number of bytes in a string.
+
+            This can be computationally expensive, only call when necessary.
+            """
+            return len(str(data).encode("utf-8"))
+
+        final_batch = []
+        row_bytes_limit = CLICKHOUSE_SINGLE_ROW_INSERT_BYTES_LIMIT
+        # Set the value byte limit to be anything over 1MiB to catch
+        # payloads with multiple large values that are still under the
+        # single row insert limit.
+        val_byte_limit = 1 * 1024 * 1024
+
+        # Call batch is bigger than the single row insert limit, could be
+        # totally fine, but make sure that no single row is over the limit
+        seen_bytes = 0
+        for i, item in enumerate(batch):
+            bytes_size = _num_bytes(item.model_dump_json())
+            seen_bytes += bytes_size
+
+            # If bytes_size > row_bytes_limit, this item is too large,
+            # iterate through the json value keys to find and replace
+            # the large values with a placeholder.
+            if bytes_size > row_bytes_limit:
+                if isinstance(item.req, tsi.CallStartReq):
+                    id_, mode = (item.req.start.id, "start")
+                    for key in item.req.start.inputs:
+                        if _num_bytes(item.req.start.inputs[key]) > val_byte_limit:
+                            item.req.start.inputs[key] = ENTITY_TOO_LARGE_ERROR
+                else:
+                    id_, mode = (item.req.end.id, "end")
+                    for key in item.req.end.output:
```

Comment:
almost (: ... output is not necessarily a dict

Reply:
ahhh